File: 4_greedy_그리디 알고리즘/5_회의실 배정(그리디)/answer2.py
import sys
sys.stdin = open("in1.txt", "r")
n = int(input())
meeting = []
for i in range(n):
    a, b = map(int, input().split())
    meeting.append((a, b))
meeting.sort(key=lambda x: (x[1], x[0]))

et = 0
cnt = 0

print(meeting)

for x, y in meeting:
    print(x, y)
    if x >= et:
        et = y
        cnt += 1
print(cnt)

File: 4_greedy_그리디 알고리즘/5_회의실 배정(그리디)/answer2.js
let fs = require("fs");
let content = fs.readFileSync("in2.txt").toString();

console.log(content)

let n;
let meetings = []

content.trim().split('\n').map((v, i) => {

    if (i == 0) {
        n = parseInt(v.trim())
    }

    else {
        meetings.push(v.trim().split(' ').map(v => Number(v)))
    }
})


// # order by y asc,a asc
// 오름 차순
meetings.sort((a, b) => a[1] == b[1] ? a[1] - b[1] : a[1] - b[1] + (a[0] - b[0]) * 0.0001)

let et = 0
let cnt = 0

console.log(meetings)

for (meet of meetings) {
    let x = meet[0]
    let y = meet[1]

    // console.log(x, y)
    if (x >= et) {
        et = y
        cnt += 1
    }
}

console.log("최대 사용 개수 :", cnt)
